zerver/management/commands/deactivate_realm.py

```diff
@@ -1,23 +1,35 @@
 from argparse import ArgumentParser
 from typing import Any
 
-from zerver.lib.actions import do_deactivate_realm
+from zerver.lib.actions import do_add_deactivated_redirect, do_deactivate_realm
 from zerver.lib.management import ZulipBaseCommand
 
 
 class Command(ZulipBaseCommand):
     help = """Script to deactivate a realm."""
 
     def add_arguments(self, parser: ArgumentParser) -> None:
+        parser.add_argument('--redirect_url', metavar='<redirect_url>',
+                            help="URL to which the realm has moved")
         self.add_realm_args(parser, True)
 
     def handle(self, *args: Any, **options: str) -> None:
         realm = self.get_realm(options)
+        redirect_url = options['redirect_url']
+
         assert realm is not None  # Should be ensured by parser
 
         if realm.deactivated:
-            print("The realm", options["realm_id"], "is already deactivated.")
+            print("The realm", options['realm_id'], "is already deactivated.")
+            if options['redirect_url']:
+                print("Setting the redirect URL to", options['redirect_url'])
+                do_add_deactivated_redirect(realm, redirect_url)
             exit(0)
+
         print("Deactivating", options["realm_id"])
         do_deactivate_realm(realm)
+
+        if options['redirect_url'] and not realm.deactivated_redirect:
```

Comment:
If we are setting the deactivate_redirect field always, we can remove the duplicated calls to `do_add_deactivated_redirect` and always call it if a redirect_url argument is supplied.

Reply:
ah yes!! doing it!